### PR TITLE
Travis CI: test with OpenJDK 8

### LINKED ISSUE #1188: Travis: test with OpenJDK

### PR DESCRIPTION
**Pull request by plata**
fixes #1188

### CODE REVIEW

**Comment by plata** [comment]
The build works but there are character encoding problems.

**Comment by madoar** [comment]
Do you have an idea how we can fix the encoding problems?

**Comment by plata** [comment]
No. I don't even have an idea why they occur in the first place.

**Comment by madoar** [comment]
@plata after a short search I've found [this](https://stackoverflow.com/questions/27931668/encoding-problems-when-running-an-app-in-docker-python-java-ruby-with-u), maybe the solution there works for us.

**Comment by madoar** [comment]
I think it would be best to specify the commands we execute in the Dockerfile.
We can then build and run the container via `docker build` and `docker run`, see https://docs.docker.com/get-started/part2/#build-the-app for more details.

**Comment by plata** [comment]
Now we get
```
package javafx.geometry does not exist
```

**Comment by madoar** [comment]
I think `openjfx` is not installed?

**Comment by plata** [comment]
You are right. Actually I was looking for it but then somehow mixed openjfx and openjdk and got confused because that's installed already :P

**Comment by plata** [comment]
It's working now.

**Comment by plata** [comment]
You cannot name jobs in Travis at the moment (see https://github.com/travis-ci/travis-ci/issues/5898).

OpenJDK 9 should just be copy+paste.

**Comment by plata** [comment]
Seems like it's possible to set the `jdk` variable (it's not used but shows up in the table correctly).

**Comment by plata** [comment]
@madoar do you have Ubuntu 18.04 installed? The OpenJDK 9 build doesn't like the `JAVA_HOME`.

**Comment by madoar** [comment]
No, I'm using Linux Mint 18.3.

**Comment by plata** [comment]
Any ideas what the problem could be?

**Comment by madoar** [comment]
Are you a hundred percent sure, the JAVA_HOME path set in the travis file is the correct one?
It may be useful to add a `ls`, just to be sure.

**Comment by plata** [comment]
Ok... Seems like there is no `openjdk-9-jdk` package.

**Comment by plata** [comment]
Don't know why it's not working.

**Comment by madoar** [comment]
Just asking:
Why don't we use an openjdk docker image? 
See https://hub.docker.com/_/openjdk/

**Comment by plata** [comment]
We can try that. Which one should I use?

**Comment by madoar** [comment]
I think we should start with `openjdk:8`.
If it works we can try `openjdk:9` and lastly `openjdk:10`.
The benefit of using these images should be, that you should not need to set the (java) paths.

**Comment by plata** [comment]
Do we need a custom docker file for that? I'm not that familiar with Docker really. I thought you were talking about an Ubuntu image containing OpenJDK 8.

**Comment by madoar** [comment]
I am talking about a docker image.
The docker image is called `openjdk`, while the tag afterwards specifies the to be used openjdk version.
So if you need an ubuntu image containing OpenJDK 8 you can use the image `openjdk:8`.

**Comment by plata** [comment]
Seems to be the same problem.

**Comment by madoar** [comment]
When using the `openjdk` image you shouldn't need to set `JAVA_HOME`.
In addition you can try removing the language parameters.

Have you tried creating and executing the containers locally?

**Comment by plata** [comment]
No, I didn't try locally.

File: .travis.yml [hunk 1]
```diff
@@ -15,6 +15,45 @@ jobs:
...
+      sudo: required
+      services:
+        - docker
+      before_install:
+        - docker run -d --name ubuntu-16-04 -v $(pwd):/travis openjdk:8 tail -f /dev/null
+      install:
+        - docker exec -t ubuntu-16-04 bash -c "apt -qq update;
+                                               apt install -y git;
+                                               apt install -y maven;
+                                               apt install -y openjdk-8-jdk;
```
Comment (madoar) [inline .travis.yml]: This line isn't needed, when using the image `openjdk:8`

File: .travis.yml [hunk 2]
```diff
@@ -15,6 +15,45 @@ jobs:
...
+      sudo: required
+      services:
+        - docker
+      before_install:
+        - docker run -d --name ubuntu-16-04 -v $(pwd):/travis openjdk:9 tail -f /dev/null
+      install:
+        - docker exec -t ubuntu-16-04 bash -c "apt -qq update;
+                                               apt install -y git;
+                                               apt install -y maven;
+                                               apt install -y openjdk-9-jdk;
```
Comment (madoar) [inline .travis.yml]: This line isn't needed when using the image `openjdk:9`

**Comment by plata** [comment]
Ok. Let's simply not support OpenJDK9 for now (this is clearly not a Phoenicis issue).

**Comment by qparis** [comment]
Anyway, JDK 10 is out. I think we should support this one because there the memory management is optimized and also we could take advantage of type hintings. We do you think?

**Comment by madoar** [comment]
@qparis the issue is not really JDK 9 or JDK 10.
The problem seems to be the way JavaFX (i.e. OpenJFX) works together with OpenJDK 9+.

As far as I understand it, up to OpenJDK 8 it was possible to overlay an OpenJFX build over an existing JDK, i.e. you could install OpenJFX after installing OpenJDK 8 on your system.

Starting with OpenJDK 9 this isn't possible anymore.
Now you need to build the (Open)JDK 9+ after installing OpenJFX.
When building OpenJDK 9+ you seem to need to set a reference to the OpenJFX installation/build. 

More details can be found [here](https://wiki.openjdk.java.net/display/OpenJFX/Building+OpenJFX#BuildingOpenJFX-IntegrationwithOpenJDK).

**Comment by plata** [comment]
To add to @madoar: the whole issue seems to be related to the module system which has been introduced. Of course, we can decide to support *Oracle* JDK 10 but that's completely unrelated to this PR.

**Pull request merged**